torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -301,6 +307,11 @@ def inner(*flat_args):
             if isinstance(o, torch.Tensor):
                 curr_storage = StorageWeakRef(o.untyped_storage())
                 out_tensor_alias_counts[curr_storage] += 1
+
+                if is_export and isinstance(o, FunctionalTensor):
+                    o_group_id = mode._partial_frozen_storage.get(curr_storage, None)
+                    if o_group_id in mode._mutated_partial_frozen_storage:
+                        raise RuntimeError("Cannot return on mutated frozen tensor")
```

Comment:
Maybe more descriptive error message would be good here: "We can return output tensor that has been mutated after aten.to method. Please consider adding .clone() to work around this issue"